languages/purescript/reference/errors/EscapedSkolem.md

```diff
@@ -0,0 +1,35 @@
+# `EscapedSkolem` Error
+
+## Example
+
+```purescript
+test = do
+  r <- runST (newSTRef 0)
+  return 0
+```
+
+## Cause
+
+Skolem constants are types which only unify with themselves, and a limited range of unification variables. They are used to check polymorphic types, usually when using Rank N Types.
+
+We say that a skolem _escapes its scope_ when an attempt is made to unify it with a unification variable which was generated outside the scope in which the skolem was generated.
```

Comment:
Maybe remove this paragraph or move it below the next paragraph. I know this is a tricky error to explain.. It might help to define unification as well, perhaps in the first paragraph:

> and a limited range of unification variables. Unification is the process of verifying that two type variables are compatible. Within `do` notation, the compiler must unify the type of each line.

Or something..

Reply:
It'd probably be helpful to include the types for `runST` and `newSTRef`, as well as an example of correct usage